### ISSUE TITLE
Custom syntax highlighting for macro forms

### ISSUE THREAD

**Issue by katox** [cursive-ide/cursive#1360]
In addition to your ideas about better error reporting it would be really useful to be able to define custom syntax highlighters for macros. It would lead to better clarity of forms that differ from standard (fn arg1 arg2 ...) form.

Relates to #791, #522, , #147, #953.

**Comment by harishtella** [comment]
Yes, I really miss this from other editors I've used, like Atom and VSCode. Here is an example.

![Screen Shot 2020-06-02 at 12 25 50 PM](https://user-images.githubusercontent.com/362292/83550304-410ee400-a4cc-11ea-96f2-d3fb626935e6.png)
